l10n_nl_tax_invoice_basis/models/account_move_line.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2017 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    l10n_nl_date_invoice = fields.Date(
+        compute="_compute_l10n_nl_date_invoice", store=True
+    )
+
+    @api.depends(
+        "move_id",
+        "move_id.invoice_date",
+        "move_id.type",
+        "company_id",
+        "company_id.l10n_nl_tax_invoice_basis",
+        "company_id.country_id",
+    )
+    def _compute_l10n_nl_date_invoice(self):
+        for line in self.filtered(lambda l: l.parent_state == "draft"):
```

Comment:
If we only calculate if parent_state is draft shouldn't we calculate the date for posted/paid invoices when installing the module? Why do we only calculate if state is draft anyhow?

Reply:
I think this check should be removed, the state isn't a factor in calculating the date.

Reply:
State is checked to prevent invoices to be retroactively updated in case company configuration is changed.